src/core/datastore/networkstore.js

```diff
@@ -88,42 +97,15 @@ export class NetworkStore {
    * @return  {Observable}                                                Observable.
    */
   find(query, options = {}) {
-    const useDeltaFetch = options.useDeltaFetch === true || this.useDeltaFetch;
-    const stream = KinveyObservable.create((observer) => {
-      // Check that the query is valid
-      if (isDefined(query) && !(query instanceof Query)) {
-        return observer.error(new KinveyError('Invalid query. It must be an instance of the Query class.'));
-      }
-
-      // Create the request
-      const config = {
-        method: RequestMethod.GET,
-        authType: AuthType.Default,
-        url: url.format({
-          protocol: this.client.apiProtocol,
-          host: this.client.apiHost,
-          pathname: this.pathname
-        }),
-        properties: options.properties,
-        query: query,
-        timeout: options.timeout,
-        client: this.client
-      };
-      let request = new KinveyRequest(config);
-
-      // Should we use delta fetch?
-      if (useDeltaFetch === true) {
-        request = new DeltaFetchRequest(config);
-      }
+    const errPromise = this._ensureValidQuery(query);
+    if (errPromise) {
+      return wrapInObservable(errPromise);
+    }
 
-      // Execute the request
-      return request.execute()
-        .then(response => response.data)
-        .then(data => observer.next(data))
-        .then(() => observer.complete())
-        .catch(error => observer.error(error));
-    });
-    return stream;
+    options.useDeltaFetch = options.useDeltaFetch || this.useDeltaFetch;
```

Comment:
I dunno, I copied it from `findById()` of the old `NetworkStore`. But it didn't make sense in a `..byId` method so I removed it from there.

Considering it now, I think the user should be able to set it on a "per-request" basis and it should take precedence over the "general" setting. I'll change it to that, you guys let me know if you'd like something else, or remember a reason for the different settings. Cause I see it in other places in `NetworkStore` (like `find()`) as `const useDeltaFetch = options.useDeltaFetch === true || this.useDeltaFetch;` (which is almost the same) and as `options = assign({ useDeltaFetch: this.useDeltaFetch }, options);` in `CacheStore.pull()` (which I think is the best).

Reply:
Actually, I've been meaning to ask, does `deltaFetch` even make sense in a `find()` method? I feel it should only be part of sync-related methods.

Reply:
Since `pull()` was essentially a special case of a `find()`, `deltaFetch` on a `find()` made sense. Some of this discussion again goes back to the discussion we had about the `CacheStore`. On `Network` and `Sync`, none of this factors in. I'd suggest leaving it in there for now.

I agree with the setting override. The `options` value should supercede the "general" setting.

Reply:
Yes it should be part of `find()`.

Reply:
I've pushed the changes for this. I feel `find` is the special case of `pull`, but this is topic for another conversation.